lib/generator.js

```diff
@@ -53,6 +54,14 @@ const shouldIgnoreDir = dirPath =>
   dirPath === '.git'
   || dirPath.startsWith(`.git${path.sep}`);
 
+const validateOptions = options => {
+  const unexpectedOptions = Object.keys(options).filter(param => !GENERATOR_OPTIONS.includes(param));
+  const readableUnexpectedOptions = unexpectedOptions.join('\', \'');
+  if (unexpectedOptions.length === 0) return '';
+  if (unexpectedOptions.length === 1) return `An unexpected param was given: '${readableUnexpectedOptions}'`;
+  return `Unexpected params were given: '${readableUnexpectedOptions}'`;
+};
+
```

Comment:
```suggestion
const validateOptions = options => {
  const unexpectedOptions = Object.keys(options).filter(param => !GENERATOR_OPTIONS.includes(param));
  if (!unexpectedOptions.length) return;
  const readableUnexpectedOptions = unexpectedOptions.join(', ');
  
  return `These options are not supported by the generator: ${readableUnexpectedOptions}`;
};
```

- when we console log other errors, we do not put them in between `'` this is why I suggest above to remove those
- 0 is always falsy so no need to do `=== 0`. You also do not have to return empty string as it is not needed anywhere
- are you sure we need 2 different log messages for plural and singular? in other places we just approach it like this `Error: This template requires the following missing params: server.` and I think that form DX perspective this is pretty fine, and as a result one line of code to maintain as then you can write it as I wrote it in my suggestion
- I proposed different error message, especially that for us `params` can be mistaken with params passed with `templateParams`. So at the end these are options

Reply:
I think that the best would be if you move this function to utils, let us try to make everything possible to not grow generator.js to much 😅

Reply:
Thanks @derberg! Just added the `return ''` to return the same type of object in any case, but it's completely understandable, for all the other cases as well.

_"are you sure we need 2 different log messages for plural and singular?"_, not really, it was just to make it more fancy 😁.

I agree about moving out this added logic to utils, so I'm gonna give it a try.